core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java

```diff
@@ -801,6 +804,12 @@ public boolean transitionToDispatching()
         return queryState.setIf(DISPATCHING, currentState -> currentState.ordinal() < DISPATCHING.ordinal());
     }
 
+    public boolean transitionToPlanning(Thread planningThread)
```

Comment:
actually, could we handle interruption in `SqlQueryExecution` (via state change listener)?